### PR TITLE
compose: add vouch to compose

### PR DESCRIPTION
**Pull request by dB2510**
Adds vouch to compose and also made some changes to beaconmock which were needed by vouch specifically.
Note: Vouch doesn't support block proposals because of RANDAO check in go-eth2-client.

category: feature
ticket: #1403

### CODE REVIEW

File: testutil/beaconmock/headproducer.go [hunk 1]
```diff
@@ -176,7 +197,28 @@ func (p *headProducer) handleEvents(w http.ResponseWriter, r *http.Request) {
 	query := r.URL.Query()
 	query.Set("stream", sseStreamID) // Add sseStreamID for sse server to serve events on.
 	r.URL.RawQuery = query.Encode()
-	p.server.ServeHTTP(w, r)
+
+	for _, topic := range query["topics"] {
```
Comment (corverroos) [inline testutil/beaconmock/headproducer.go:201]: this will break if multiple topics requested

Comment (dB2510) [reply testutil/beaconmock/headproducer.go:201]: yeah initially i tried with single server that handles different topics but on the vouch side it has 2 different handlers for head and block events and it panics if head event handler gets block event and vice versa: https://github.com/attestantio/vouch/blob/9b8114dc0374d7e5bb6b01ce5f519213bfe30d37/services/cache/standard/service.go#L77

Here's the head event handler: https://github.com/attestantio/vouch/blob/9b8114dc0374d7e5bb6b01ce5f519213bfe30d37/services/cache/standard/events.go#L39

Here's the block event handler: https://github.com/attestantio/vouch/blob/9b8114dc0374d7e5bb6b01ce5f519213bfe30d37/services/cache/standard/events.go#L27

File: testutil/beaconmock/headproducer.go [hunk 2]
```diff
@@ -95,20 +96,55 @@ func (p *headProducer) setCurrentHead(currentHead *eth2v1.HeadEvent) {
...
-		Event: []byte("head"),
-		Data:  data,
-	})
+	blockData, err := json.Marshal(currentBlock)
+	if err != nil {
+		panic(err) // This should never happen and this is test code sorry ;)
+	}
+
+	// Publish head events.
+	for _, streamID := range p.getStreamIDs("head") {
```
Comment (corverroos) [inline testutil/beaconmock/headproducer.go:134]: make these constants `topicHead` and `topicBlock`

File: testutil/beaconmock/headproducer.go [hunk 3]
```diff
@@ -173,9 +209,33 @@ func (p *headProducer) handleGetBlockRoot(w http.ResponseWriter, r *http.Request
...
+	p.server.CreateStream(streamID)
+
 	query := r.URL.Query()
-	query.Set("stream", sseStreamID) // Add sseStreamID for sse server to serve events on.
+	query.Set("stream", streamID) // Add sseStreamID for sse server to serve events on.
 	r.URL.RawQuery = query.Encode()
+
+	for _, topic := range query["topics"] {
+		if topic != "head" && topic != "block" {
+			log.Warn(context.Background(), "Unknown topic requested", nil, z.Str("topic", topic))
```
Comment (corverroos) [inline testutil/beaconmock/headproducer.go:222]: Unsupported not unknown

**Review by corverroos — APPROVED** [review]

File: testutil/beaconmock/headproducer.go [hunk 4]
```diff
@@ -173,9 +209,33 @@ func (p *headProducer) handleGetBlockRoot(w http.ResponseWriter, r *http.Request
 
 // handleEvents is a http handler to handle "/eth/v1/events".
 func (p *headProducer) handleEvents(w http.ResponseWriter, r *http.Request) {
```
[the remark this answers is not in the document]
Comment (dB2510) [reply testutil/beaconmock/headproducer.go:216]: Done!

**Review by corverroos — APPROVED** [review]

**Pull request merged**